gremlin-go/driver/client.go

```diff
@@ -33,16 +34,16 @@ type ClientSettings struct {
 
 // Client is used to connect and interact with a Gremlin-supported server.
 type Client struct {
-	host            string
-	port            int
+	url             string
+	authInfo        *AuthInfo
+	tlsConfig       *tls.Config
 	logHandler      *logHandler
 	transporterType TransporterType
 	connection      *connection
 }
 
-// NewClient creates a Client, configures it with the given parameters and creates connection.
-// Return a non-nil error if creating this connection fails.
-func NewClient(host string, port int, configurations ...func(settings *ClientSettings)) (*Client, error) {
+// NewClient creates a Client and configures it with the given parameters.
+func NewClient(url string, authInfo *AuthInfo, tlsConfig *tls.Config, configurations ...func(settings *ClientSettings)) (*Client, error) {
```

Comment:
I think all these configurations need a default and be moved to the `configurations`. As we currently have it, it becomes mandatory.